### DISCUSSION TITLE
How to change to the accuracy of the logical date from second to millisecond

### DISCUSSION THREAD

**Discussion by bianbianzhu** [apache/airflow#22755] [Q&A] [answered]
I am building a pipeline using S3 > Sqs > lambda > airflow (mwaa).

Mwaa's logical date has an accuracy of second. Apparently, the lambda will be triggered multiple times in 1s and thus results in the error: duplicate key value violates unique constraint in airflow. 

Is there anyway I can change the meta database's schema regarding the logical date (execution date) to millisecond?

Really appreciate your ideas.

**Comment by potiuk** [comment, marked as answer]
In Airflow 2.2+ the primary key is dag_run_id not a logical date any more.  Migrate to to 2.2 and you will be able to make use of it.

**Reply by bianbianzhu** [reply]
Thanks @potiuk  that is a good one. Just we are using MWAA and the version is 2.0.2 and it is going to be a process if we migrate version, wondering if we can do anything on the existing version.

There is a configuration option https://airflow.apache.org/docs/apache-airflow/2.0.2/configurations-ref.html#sql-alchemy-schema and we are now looking at if can possibly update the schema column execution date precision to millisecond.

**Reply by potiuk** [reply]
I believe it is disabled on MWAA and your only option is to migrate.
